spec/compiler/codegen/macro_spec.cr

```diff
@@ -1876,4 +1876,29 @@ describe "Code gen: macro" do
       foo
     )).to_string.should eq("  42")
   end
+
+  it "access to the program with @top_level" do
+    run(%(
+      class Foo
+        def bar
+          {{@top_level.name.stringify}}
+        end
+      end
+
+      Foo.new.bar
+      )).to_string.should eq("main")
```

Comment:
This is not directly related, but it feels a bit odd that the top-level scope is called `main`. `top_level` would probably be more appropriate. But I', not sure what it should be. 
`@top_level.stringify` is `<Program>`, btw.

Reply:
top_level sounds good. TBH this was a test "after the fact", and I felt that naming it differently was another issue. Oddly enough, I couldn't find where that comes from...

Reply:
https://github.com/crystal-lang/crystal/blob/421fa11bf05b11441b74bbeb2762b673f96fec3f/src/compiler/crystal/program.cr#L126

It should be fine to change it, it was never shown until now.

Reply:
I took the liberty to add the change in this PR. Thanks @asterite !